app/models/location.rb

```diff
@@ -5,6 +5,7 @@ class Location < ApplicationRecord
   validates :time_zone, presence: true
 
   has_many :announcements, inverse_of: :location, dependent: :destroy
+  has_many :widgets, inverse_of: :location, dependent: :destroy
```

Comment:
not sure I understand why locations own widgets, I imagined the location the client wanted would only have to live on the client in a param

Reply:
I think because different locations have different data for the weather and for the announcements specifically, so when we do a single query for every widget we want to get only the relevant information for the location running the app. The primary location is in the env variable and then the query would just be something like the screenshot: `{primaryLocation{widgets{...}}` and then no other location would need to be specified. Eventually, we could add another query that would be something like: `{locations(city_name: "Boulder"){widgets{ ... }}}` if the location wanted is not the primary.

Reply:
I guess what I assumed was that the widgets would be the same for all locations. The content would change based on a query param that would be passed into the weather widget.

Reply:
the way you have it should be fine though.